libs/src/oracle-sdk-v2/services/oraclev1/ethers/factory.ts

```diff
@@ -0,0 +1,152 @@
+
+    const result: SharedRequest = {
+      id,
+      chainId,
+      oracleAddress,
+      oracleType,
+      requester: assertAddress(requester),
+      identifier,
+      time: timestamp.toString(),
+      ancillaryData,
+    };
+    if (currency) result.currency = assertAddress(currency);
+    if (reward) result.reward = reward;
+    if (finalFee) result.finalFee = finalFee;
+    if (proposer) result.proposer = proposer;
+    if (proposedPrice) result.proposedPrice = proposedPrice;
+    if (expirationTime)
+      result.proposalExpirationTimestamp = expirationTime.toString();
+    if (disputer) result.disputer = disputer;
+    if (price) result.settlementPrice = price;
+    if (reward) result.settlementPayout = reward;
+    if (settleTx) result.settlementHash = settleTx;
+    if (requestBlockNumber)
+      result.requestBlockNumber = requestBlockNumber.toString();
+    if (requestTx) result.requestHash = requestTx;
+    if (price) result.settlementPrice = price;
+    if (reward) result.settlementPayout = reward;
+    if (settleTx) result.settlementHash = settleTx;
+    if (requestBlockNumber)
+      result.requestBlockNumber = requestBlockNumber.toString();
+    if (requestTx) result.requestHash = requestTx;
+    if (state) result.state = convertToSharedState(state);
+    if (proposeTx) result.proposalHash = proposeTx;
+    if (disputeTx) result.disputeHash = disputeTx;
+    if (proposeBlockNumber)
+      result.proposalBlockNumber = proposeBlockNumber.toString();
+    if (disputeBlockNumber)
+      result.disputeBlockNumber = disputeBlockNumber.toString();
+    if (requestLogIndex) result.requestLogIndex = requestLogIndex.toString();
+    if (proposeLogIndex) result.proposalLogIndex = proposeLogIndex.toString();
+    if (disputeLogIndex) result.disputeLogIndex = disputeLogIndex.toString();
+    if (settleBlockNumber)
+      result.settlementBlockNumber = settleBlockNumber.toString();
+    if (settleLogIndex) result.settlementLogIndex = settleLogIndex.toString();
+
+    // unable to get the following values, omit their keys to avoid
+    // overriding other sources when merged in final store
+    // dont know how this comes from events
+    // settlementRecipient: null
+    // settlementTimestamp: null,
+    // requestTimestamp: null,
+
+    return result;
+  };
+export type Api = {
+  updateFromTransactionReceipt: (receipt: TransactionReceipt) => void;
+};
+export const Factory = (config: Config): [ServiceFactory, Api] => {
+  const convertToSharedRequest = ConvertToSharedRequest(
+    config.chainId,
+    assertAddress(config.address),
+    "Optimistic Oracle V1"
+  );
+  const provider = new ethers.providers.JsonRpcProvider(config.url);
+  const oo = new OptimisticOracle(provider, config.address, config.chainId);
+  const events = new Events();
+  function updateFromTransactionReceipt(receipt: TransactionReceipt) {
+    oo.updateFromTransactionReceipt(receipt);
+    const requests: Request[] = oo.listRequests();
```

Comment:
this re-uses logic from the sdk-v1, where the oracle client already had ability to decode receipts and update request data. we update the class and fetch all known requests